src/main/java/picard/util/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,260 @@
+ */
+
+public final class AlleleSubsettingUtils {
+    private AlleleSubsettingUtils() {
+    }  // prevent instantiation
+
+    @Deprecated // use {@link Allele.NON_REF} (once released and htsjdk is properly reved in picard
+    public static Allele NON_REF_ALLELE = Allele.create("<NON_REF>");
+
+    /**
+     * Method to subset the alleles in the VariantContext to those in the input snp.
+     * Will also subset appropriately the AD and PL fields in all the genotypes, and in
+     * the case of a monomorphic variant with a NON_REF allele, will replace the NON_REF
+     * allele with the non-reference allele from snp.
+     * <p>
+     * returns null if it is not possible to subset ctx to the alleles in snp.
+     *
+     * @param ctx The VariantContext to subset.
+     * @param snp Snp whose alleles are used for subsetting ctx.
+     * @return a VariantContext subsetted to the alleles in snp, or null if unable.
+     */
+    public static VariantContext subsetVCToMatchSnp(final VariantContext ctx, final Snp snp) {
+        if (ctx.isFiltered()) return null;
+
+        // we can use this VC if it contains both alleles in snp as alleles, one of which is the reference,
+        // or if it contains the NON_REF_ALLELE and one of the snp's alleles, and its reference allele is length 1.
+
+        //TODO: figure out a way to identify the allele corresponding to the snp in the presence of a deletion
+
+        if (ctx.getReference().length() != 1) return null;
+
+        // one of the alleles in snp must match the reference allele
+        final Optional<Byte> referenceAlleleMaybe = Stream.of(snp.getAllele1(), snp.getAllele2())
+                .filter(b -> b == ctx.getReference().getBases()[0]).findAny();
+        if (!referenceAlleleMaybe.isPresent()) return null;
+
+        final byte refAllele = referenceAlleleMaybe.get();
+        final byte otherAllele = snp.getAllele1() == refAllele ? snp.getAllele2() : snp.getAllele1();
+
+        // do we have otherAllele in ctx?
+        final Optional<Allele> altAlleleMaybe = ctx.getAlternateAlleles()
+                .stream()
+                .filter(a -> a.length() == 1 && a.getBases()[0] == otherAllele)
+                .findAny();
+
+        if (altAlleleMaybe.isPresent()) {
+            if (ctx.getAlleles().size() == 2) return ctx;
+
+            return AlleleSubsettingUtils.subsetAlleles(ctx, Arrays.asList(ctx.getReference(), altAlleleMaybe.get()));
+        }
+
+        // if not, perhaps we have NON_REF_ALLELE
+        final Optional<Allele> nonRefAlleleMaybe = ctx.getAlternateAlleles().stream().filter(a -> a.equals(NON_REF_ALLELE)).findAny();
+        if (nonRefAlleleMaybe.isPresent()) {
+
+            final VariantContext vcSubsetted = ctx.getAlleles().size() == 2 ? ctx : AlleleSubsettingUtils.subsetAlleles(ctx, Arrays.asList(ctx.getReference(), nonRefAlleleMaybe.get()));
+            return AlleleSubsettingUtils.swapAlleles(vcSubsetted, NON_REF_ALLELE, Allele.create(otherAllele));
```

Comment:
the fingerprinting-code  will compare the two alleles in the VC to those in the haplotype block, but it doesn't know how to deal with multiallelic sites, or Non-ref. if the alleles in the VC do not match those in the haplotype block, it will not modify the fingerprint and continue.

If there's only one allele+<NON_REF> I want it to be a Homozygous site which is what this does. if there was even the slightest indication of a third allele, it would show up in the VC...so this is just a way to make sure that I can use the PLs

Reply:
OK?

Reply:
Great, thanks for the explanation.